mito-ai/src/Extensions/AiChat/ChatTaskpane.tsx

```diff
@@ -901,23 +946,57 @@ const ChatTaskpane: React.FC<IChatTaskpaneProps> = ({
         setChatHistoryManager(chatHistoryManager)
     }
 
-    const markAgentForStopping = (): void => {
-        // Signal that the agent should stop after current task
+    const markAgentForStopping = async (): Promise<void> => {
+        // 1. Immediately abort any ongoing requests
+        if (activeRequestControllerRef.current) {
+            activeRequestControllerRef.current.abort();
+            activeRequestControllerRef.current = null;
+        }
+        
+        // 2. Disconnect any active stream handlers
+        if (streamHandlerRef.current) {
+            websocketClient.stream.disconnect(streamHandlerRef.current, null);
+            streamHandlerRef.current = null;
+        }
+        
+        // 3. Clear streaming content
+        streamingContentRef.current = '';
```

Comment:
The streamignContentRef is only used for chat mode, right? In agent mode we don't do any streaming. So does it make sense to do something with that here?

Reply:
Same with the streamHanlderRef?